controllers/common/types.go

```diff
@@ -65,31 +65,30 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			r.Log.Error(err, "failed to recover chaos")
 			return ctrl.Result{Requeue: true}, nil
 		}
-		return ctrl.Result{}, nil
-	}
+	} else {
+		// Start failure action
+		r.Log.Info("Performing Action")
 
-	// Start failure action
-	r.Log.Info("Performing Action")
+		status := chaos.GetStatus()
 
-	status := chaos.GetStatus()
+		err = r.Apply(ctx, req, chaos)
+		if err != nil {
+			r.Log.Error(err, "failed to apply chaos action")
 
-	err = r.Apply(ctx, req, chaos)
-	if err != nil {
-		r.Log.Error(err, "failed to apply chaos action")
+			updateError := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
If `apply action` is failed, we should probably not update the chaos object and recover it. @YangKeao what do you think? Maybe we need a separate pr to fix this issue.